cupyx/scipy/sparse/linalg/_eigen.py

```diff
@@ -79,8 +81,11 @@ def eigsh(a, k=6, *, which='LM', ncv=None, maxiter=None, tol=0,
     V = cupy.empty((ncv, n), dtype=a.dtype)
 
     # Set initial vector
-    u = cupy.random.random((n,)).astype(a.dtype)
-    V[0] = u / cublas.nrm2(u)
+    if v0 is None:
+        u = cupy.random.random((n,)).astype(a.dtype)
+        V[0] = u / cublas.nrm2(u)
+    else:
+        V[0] = cupy.asarray(v0)
```

Comment:
In CuPy, ndarray arguments must always be `cupy.ndarray` to avoid implicit host-to-device copy. So could you remove `cupy.asarray` here?